cmake/modules/FindHOST_TOOLS.cmake

```diff
@@ -62,22 +62,24 @@ find_program(BOSSAC bossac)
 find_program(IMGTOOL imgtool)
 
 # Pick host system's toolchain if we are targeting posix
+zephyr_get(ZEPHYR_TOOLCHAIN_VARIANT)
 if("${ARCH}" STREQUAL "posix")
   if(NOT "${ZEPHYR_TOOLCHAIN_VARIANT}" STREQUAL "llvm")
-    set(ZEPHYR_TOOLCHAIN_VARIANT "host")
+    set(ZEPHYR_TOOLCHAIN_VARIANT "host" CACHE STRING "Zephyr toolchain variant" FORCE)
   endif()
 endif()
 
 # Keep XCC_USE_CLANG behaviour for a while.
 if ("${ZEPHYR_TOOLCHAIN_VARIANT}" STREQUAL "xcc"
     AND "$ENV{XCC_USE_CLANG}" STREQUAL "1")
-  set(ZEPHYR_TOOLCHAIN_VARIANT xcc-clang)
+  set(ZEPHYR_TOOLCHAIN_VARIANT xcc-clang CACHE STRING "Zephyr toolchain variant" FORCE)
   message(STATUS "XCC_USE_CLANG is deprecated. Please set ZEPHYR_TOOLCHAIN_VARIANT to 'xcc-clang'")
 endif()
 
 if(NOT ZEPHYR_TOOLCHAIN_VARIANT AND
    (CROSS_COMPILE OR (DEFINED ENV{CROSS_COMPILE})))
     set(ZEPHYR_TOOLCHAIN_VARIANT cross-compile)
+    set(ZEPHYR_TOOLCHAIN_VARIANT cross-compile CACHE STRING "Zephyr toolchain variant" FORCE)
```

Comment:
why do we need both?

Reply:
should not be needed, thanks for spotting.

Reply:
fixed